userspace/libsinsp/cri.cpp

```diff
@@ -400,22 +401,64 @@ bool cri_interface::is_pod_sandbox(const std::string &container_id)
 	return status.ok();
 }
 
-uint32_t cri_interface::get_pod_sandbox_ip(const std::string &pod_sandbox_id)
+void cri_interface::get_pod_info_cniresult(runtime::v1alpha2::PodSandboxStatusResponse &resp, std::string &cniresult)
+{
+	Json::Value root;
+	Json::Reader reader;
+	const auto &info_it = resp.info();
+	if(reader.parse(info_it.find("info")->second, root))
+	{
+		Json::Value jvalue;
+		if (m_cri_runtime_type == CT_CONTAINERD)
```

Comment:
Is there any chance we could *not* switch based on our perceived runtime type? E.g. just try both approaches and choose the first that returns valid data? I have two reasons:
- CT_CRI runtimes (which are neither cri-o nor containerd but still support the CRI API) can do whatever they please and just might implement one response or the other (and then would Just Work™)
- future cri-o/containerd versions might change their behavior (if e.g. `cniResult.Interfaces` becomes a standard or something)

Reply:
Smart suggestion, thanks! Changed it.